esprima.js

```diff
@@ -3616,6 +3692,10 @@
         startToken = lookahead;
         token = lookahead;
 
+        if (matchKeyword('yield')) {
```

Comment:
> Do we allow yield to be an identifier?

Probably not. Is there any potential compatibility issue, e.g. old code using `yield` as variable name?